src/Draco.Compiler.Tests/Syntax/ParserTests.cs

```diff
@@ -114,4 +220,1285 @@ public void TestEmptyFuncWithoutClosingCurly()
+            this.N<StringPart.Interpolation>();
+            {
+                this.T(TokenType.InterpolationStart);
+                this.N<Expr.String>();
+                {
+                    this.T(TokenType.LineStringStart);
+                    this.StringContent("World!");
+                    this.T(TokenType.LineStringEnd);
+                }
+            }
+            //this.StringNewline();
+            this.MissingT(TokenType.InterpolationEnd);
+            this.T(TokenType.MultiLineStringEnd);
+        }
+        var quotes = "\"\"\"";
+        this.StringTestsPlaceHolder($$"""
+            {{quotes}}
+            Hello, 
+            \{"World!"
+            {{quotes}}
+            """, UnclosedString);
+    }
+
+    [Fact]
+    public void TestVariableDeclarationWithNoTypeAndWithValueAndMissingSemicolon()
+    {
+        void Declaration()
+        {
+            this.N<Stmt.Decl>();
+            {
+                this.N<Decl.Variable>();
+                {
+                    this.T(TokenType.KeywordVal);
+                    this.T(TokenType.Identifier);
+                    this.N<ValueInitializer>();
+                    {
+                        this.T(TokenType.Assign);
+                        this.N<Expr.Literal>();
+                        {
+                            this.T(TokenType.LiteralInteger);
+                        }
+                    }
+                    this.MissingT(TokenType.Semicolon);
+                }
+            }
+        }
+        this.MainFunctionPlaceHolder("val x = 5", Declaration);
+    }
+
+    [Fact]
+    public void TestVariableDeclarationWithNoTypeAndWithNoValue()
+    {
+        void Declaration()
+        {
+            this.N<Stmt.Decl>();
+            {
+                this.N<Decl.Variable>();
+                {
+                    this.T(TokenType.KeywordVal);
+                    this.T(TokenType.Identifier);
+                    this.T(TokenType.Semicolon);
+                }
+            }
+        }
+        this.MainFunctionPlaceHolder("val x;", Declaration);
+    }
+
+    [Fact]
+    public void TestVariableDeclarationWithNoTypeAndWithValue()
+    {
+        void Declaration()
+        {
+            this.N<Stmt.Decl>();
+            {
+                this.N<Decl.Variable>();
+                {
+                    this.T(TokenType.KeywordVal);
+                    this.T(TokenType.Identifier);
+                    this.N<ValueInitializer>();
+                    {
+                        this.T(TokenType.Assign);
+                        this.N<Expr.Literal>();
+                        {
+                            this.T(TokenType.LiteralInteger);
+                        }
+                    }
+                    this.T(TokenType.Semicolon);
+                }
+            }
+        }
+        this.MainFunctionPlaceHolder("val x = 5;", Declaration);
+    }
+
+    [Fact]
+    public void TestVariableDeclarationWithNoTypeAndWithValueAndMissingValue()
```

Comment:
`WithValueAndMissingValue` is a bit confusing, `WithMissingValue` is enough to infer that it should be there but it's missing.